### PR TITLE
Add lecture series actions & new lecture list design

### LINKED ISSUE #303: Add Lecture-Series Actions
Lectures can be created in series, they are tagged with the field `SeriesIdentifier`. There should be actions like `delete-all` or `change-all` to change all Lectures inside the series.

### PR DESCRIPTION
**Pull request by mono424**
A new design of the lecture list and lecture series edit, to change a whole series at once.

### CODE REVIEW

File: web/template/admin/admin_tabs/edit-course.gohtml [hunk 1]
```diff
@@ -84,324 +84,417 @@
...
+                                                            x-ref="panel"
+                                                            x-show="moreDropdownOpen"
+                                                            x-transition.origin.top.right
+                                                            x-on:click.outside="closeMoreDropdown($refs.button)"
+                                                            :id="$id('more-dropdown-button')"
+                                                            style="display: none;"
+                                                            class="absolute left-0 mt-2 w-36 bg-white border border-black rounded shadow-md overflow-hidden"
+                                                    >
+                                                        <div>
+                                                            <button @click="lecture.startSingleEdit(); closeMoreDropdown();" class="block w-full px-4 py-2 text-left text-sm text-gray-400 dark:hover:text-gray-500 hover:text-gray-300" >
```
Comment (MatthiasReumann) [inline web/template/admin/admin_tabs/edit-course.gohtml]: The dropdown misses some `dark:` classes.

File: api/courses.go [hunk 2]
```diff
@@ -39,7 +39,9 @@ func configGinCourseRouter(router *gin.Engine) {
 	adminOfCourseGroup.POST("/presets", updatePresets)
 	adminOfCourseGroup.POST("/deleteLectures", deleteLectures)
 	adminOfCourseGroup.POST("/renameLecture/:streamID", renameLecture)
+	adminOfCourseGroup.POST("/updateLectureSeries/:streamID", updateLectureSeries)
 	adminOfCourseGroup.POST("/updateDescription/:streamID", updateDescription)
+	adminOfCourseGroup.POST("/deleteLectureSeries/:streamID", deleteLectureSeries)
```
Comment (MatthiasReumann) [inline api/courses.go]: It doesn't change behaviour, however, I think an update should be the HTTP Method PUT, and a delete DELETE. The URL the could be something like `/lectureSeries/:streamID`.

**Review by MatthiasReumann — COMMENTED** [review]
The new list looks insanely good and series updates work quite nicely, 💯 

I added a few small suggestions.

**Review by joschahenningsen — COMMENTED** [review]
This really looks neat! :fire: The only issue I'm currently seeing is that stream server and key should not be shown if the lecture hall is set :)

File: web/template/admin/admin_tabs/edit-course.gohtml (hunk 1 above)
Comment (mono424) [reply web/template/admin/admin_tabs/edit-course.gohtml]: Thank you! Added a dark background for dark mode :)

File: api/courses.go (hunk 2 above)
Comment (mono424) [reply api/courses.go]: yes thats better ! 👍

**Comment by mono424** [comment]
@joschahenningsen 
Thank you! :)
Should it be displayed if self streaming is selected ?

**Comment by joschahenningsen** [comment]
Yes :) the key and server are what self-streamers type into OBS to stream

**Comment by joschahenningsen** [comment]
I'll take care of the merge conflict and split this up in a few files for better maintainability, looks great! Thanks @mono424 :)

**Review by joschahenningsen — APPROVED** [review]
:tada:

**Pull request merged**